src/Qowaiv.Validation.Abstractions/Diagnostics/ActFailed.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace Qowaiv.Validation.Abstractions.Diagnostics;
+
+[DebuggerDisplay("Count = {Count}")]
+[DebuggerTypeProxy(typeof(CollectionDebugView))]
+internal sealed class ActFailed : IReadOnlyCollection<StackFrame>, IValidationMessage
+{
+    /// <summary>Initiates a new instance of the <see cref="ActFailed"/> class.</summary>
+    private ActFailed(StackTrace trace) => Trace = trace;
+
+    /// <inheritdoc />
+    public ValidationSeverity Severity
+        => Trace is { }
+        ? ValidationSeverity.Error
```

Comment:
I think it would perhaps be beneficial to introduce a new severity for this (e.g. `Debug`)

Reply:
Given: ``` C#
/// <summary>The validation severity of a <see cref="IValidationMessage"/>.</summary>
public enum ValidationSeverity
{
    /// <summary>Successful message (so no severity).</summary>
    None = -1,

    /// <summary>Informative message severity.</summary>
    Info = 0,

    /// <summary>Warning message severity.</summary>
    Warning = 1,

    /// <summary>Error message severity.</summary>
    Error = 2,
}
```
That can be challenging. I already considered Info a better severity, but a new level might be problematic.

Reply:
I'm not entirely sure how this snippet is meant to show that adding a new severity is challenging? If you mean that the integer values of the enum are meant to also bear some meaning, then I would propose to make `Debug` have the value of `-2`, since the message is theoretically only present during debugging and is therefore similar to `None` (except that it doesn't get filtered).